eldritch/test_events.py

```diff
@@ -262,6 +296,151 @@ def testMoveMultipleSpaces(self):
+
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=3)):
+      next_choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(sorted(next_choice.choices), ["Fight", "Flee"])
+    next_choice.resolve(self.state, "Flee")
+
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      movement = self.resolve_to_choice(CityMovement)
+
+    self.assertFalse(movement.choices)
+    movement.resolve(self.state, movement.none_choice)
+    self.resolve_until_done()
+
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Rivertown")
+    self.assertEqual(self.char.movement_points, 0)
+    # self.assertIn(zombie, self.char.possessions)
+    # TODO: take the monster as a trophy
+
+  def testMoveMultipleThroughTwoMonstersFailedEvade(self):
+    self.char.speed_sneak_slider = 1
+    monster1 = next(monster for monster in self.state.monsters if monster.name == "Maniac")
+    monster1.place = self.state.places["Easttown"]
+    monster2 = monsters.Zombie()
+    self.state.monsters.append(monster2)
+    monster2.place = self.state.places["Easttown"]
+
+    self.advance_turn(self.state.turn_number, "movement")
+    self.assertEqual(self.char.stamina, 5)
+    self.assertEqual(self.char.sanity, 5)
+
+    movement = self.resolve_to_choice(CityMovement)
+    self.assertIn("Easttown", movement.choices)
+    movement.resolve(self.state, "Easttown")
+    self.assertEqual(self.char.movement_points, 2)
+
+    movement = self.resolve_to_choice(CityMovement)
+    self.assertIn("Rivertown", movement.choices)
+    movement.resolve(self.state, "Rivertown")
+    choice = self.resolve_to_choice(MultipleChoice)
+    # TODO: Choose which one to evade first
+    # For now, Zombie is the first
+    self.assertEqual(choice.choices, ["Fight", "Evade"])
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=4)):
+      choice.resolve(self.state, "Evade")
+      next_choice = self.resolve_to_choice(MultipleChoice)
+      self.assertEqual(sorted(next_choice.choices), ["Fight", "Flee"])
+      self.assertEqual(self.char.stamina, 4)
+      self.assertEqual(self.char.sanity, 5)
+    next_choice.resolve(self.state, "Flee")
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      maniac_choice = self.resolve_to_choice(MultipleChoice)
+      self.assertEqual(maniac_choice.choices, ["Fight", "Evade"])
+      maniac_choice.resolve(self.state, "Evade")
+      movement = self.resolve_to_choice(CityMovement)
+
+    self.assertFalse(movement.choices)
+    movement.resolve(self.state, movement.none_choice)
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Easttown")
+    self.assertEqual(self.char.stamina, 4)
+    self.assertEqual(self.char.sanity, 5)
+    self.assertEqual(self.char.movement_points, 0)
+    # TODO: ^ should this pass?
+
```

Comment:
Let's add a TODO for one more test - if the character has a motorcycle, they should not be able to exhaust the motorcycle to get back any movement points after entering combat. I haven't thought too deeply about how to do that yet - maybe set movement points to -float("inf") instead of 0?

Reply:
Okay, one more TODO:
a test where you fight a dream flier in a location with a gate, lose, get pulled through the gate, immediately cast Find Gate, return, and then don't have to fight again.